.github/workflows/mvn-verify-check.yml

```diff
@@ -117,6 +117,14 @@ jobs:
           env | grep JAVA
           java -version && mvn --version && echo "ENV JAVA_HOME: $JAVA_HOME, PATH: $PATH"
 
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/**/pom.xml') }}
```

Comment:
Why do we want cache misses if there is a pom file change? 

If Maven still pulls new dependencies om top of stale cache, what does it help with?

On the other hand, we should make sure that spark-rapids-jni and the private SNAPSHOT jars are not cached longer than a day. Is there some TTL we can set, or should we make the day-precision date part of the key for SNAPSHOT versions.

Reply:
> Why do we want cache misses if there is a pom file change?

The cache action will compare local cached repository and pom, it will download dependencies not existed at local, and cache them at post stages. pom file change should not be problem.

> If Maven still pulls new dependencies om top of stale cache, what does it help with?

Many dependencies are defined as dynamic version. If the version is different from cached one, it may download latest version. If the version is same to cached one, but the jar content updated, I don't think the cache can pull latest jar. Unfortunately I didn't find TTL related args for cache action. Or we can run more tests to verify the behaviour of cache action.

Reply:
My point is that if we have hash of pom files as a caching key, and my PR adds some minor change to a pom, the cached version of the  m2 repo will not be used.

Reply:
I compared the log of with and without cache action. Only few dependencies are cached, it may not really helpful for our case.